mesheryctl/internal/cli/root/registry/generate.go

```diff
@@ -67,6 +67,35 @@ mesheryctl registry generate --directory <DIRECTORY_PATH>
 	},
 
 	RunE: func(cmd *cobra.Command, args []string) error {
+		// Check if both "spreadsheet-id" and "spreadsheet-cred" are provided together
+		if (spreadsheeetID == "" && spreadsheeetCred != "") || (spreadsheeetID != "" && spreadsheeetCred == "") {
+			return fmt.Errorf("both 'spreadsheet-id' and 'spreadsheet-cred' must be provided together")
+		}
+
+		// Check if both "registrant-def" and "registrant-cred" are provided together
+		if (pathToRegistrantConnDefinition == "" && pathToRegistrantCredDefinition != "") ||
+			(pathToRegistrantConnDefinition != "" && pathToRegistrantCredDefinition == "") {
+			return fmt.Errorf("both 'registrant-def' and 'registrant-cred' must be provided together")
+		}
+
+		// Check mutually exclusive flags
+		if spreadsheeetID != "" && pathToRegistrantConnDefinition != "" {
+			return fmt.Errorf("'spreadsheet-id' and 'registrant-def' cannot be used together")
+		}
+
+		// Check mutually exclusive flags
+		if spreadsheeetCred != "" && pathToRegistrantCredDefinition != "" {
+			return fmt.Errorf("'spreadsheet-cred' and 'registrant-cred' cannot be used together")
+		}
+
+		if csvDirectory != "" && (spreadsheeetID != "" || pathToRegistrantConnDefinition != "") {
```

Comment:
can define cobra directive for csv directory condition check in the `inti()` function like so:
```go
    generateCmd.MarkFlagsMutuallyExclusive("directory", "spreadsheet-id")
    generateCmd.MarkFlagsMutuallyExclusive("directory", "registrant-def")
```